.github/workflows/deploy.yml

```diff
@@ -8,37 +8,119 @@ on:
     branches: [ master ]
 
 jobs:
-  deploy:
-    name: Deploy Demo App
+  deploy_chime_alpha_demo:
+    name: Alpha - Deploy Meeting Demo And Meeting Readiness Checker
     runs-on: ubuntu-latest
     env:
-        GAMMA_AWS_ACCESS_KEY_ID: ${{secrets.GAMMA_AWS_ACCESS_KEY}}
-        GAMMA_AWS_SECRET_ACCESS_KEY: ${{secrets.GAMMA_AWS_SECRET_ACCESS_KEY}}
-        BETA_AWS_ACCESS_KEY_ID: ${{secrets.BETA_AWS_ACCESS_KEY}}
-        BETA_AWS_SECRET_ACCESS_KEY: ${{secrets.BETA_AWS_SECRET_ACCESS_KEY}}
         AWS_DEFAULT_REGION: us-east-1
         AWS_DEFAULT_OUTPUT: text
+        NAME: ChimeAlpha
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - name: Checkout Package
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install SAM CLI
+        run: script/github-action-awscli-installation
+      - name: Run Deployment Script
+        run: script/deploy-canary-demo
```

Comment:
I found running `script/deploy-canary-demo` under each job and then handling a switch based on `NAME` environment variable inside this script seems counter intuitive to me.

Would it be better to just drop that script file and run the commands as part of the `run` itself?

For example:
```
name: Deploy Demo App Workflow

on:
  push:
    branches: [ master ]

jobs:
  deploy_chime_alpha_demo:
    name: Alpha - Deploy Meeting Demo And Meeting Readiness Checker
    runs-on: ubuntu-latest
    env:
        AWS_DEFAULT_REGION: us-east-1
        AWS_DEFAULT_OUTPUT: text
        NAME: ChimeAlpha
    steps:
      - name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          aws-region: us-east-1
      - name: Checkout Package
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - name: Install SAM CLI
        run: script/github-action-awscli-installation
      - name: Run Deployment Script
        run:  |
          echo "Deploying to alpha stage for canary"
          npm run deploy -- -b chime-sdk-demo-canary -s chime-sdk-demo-canary -o chime-sdk-demo-canary -u false -i eu-south-1 -t -l
          npm run deploy -- -b chime-sdk-meeting-readiness-checker-dev-canary -s chime-sdk-meeting-readiness-checker-dev-canary -a meetingReadinessChecker -u false -t -l
  
  deploy_chime_beta_demo:
    name: Beta - Deploy the Serverless Meeting and Meeting Readiness Checker Demos that talks to Chime Client
    runs-on: ubuntu-latest
    env:
        AWS_DEFAULT_REGION: us-east-1
        AWS_DEFAULT_OUTPUT: text
        NAME: ChimeBeta
        GAMMA_CHIME_ENDPOINT: ${{secrets.GAMMA_CHIME_ENDPOINT}}
        GAMMA_CHIME_SERVICE_PRINCIPAL: ${{secrets.GAMMA_CHIME_SERVICE_PRINCIPAL}}
    steps:
      - name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.BETA_AWS_ACCESS_KEY }}
          aws-secret-access-key: ${{ secrets.BETA_AWS_SECRET_ACCESS_KEY }}
          aws-region: us-east-1
      - name: Checkout Package
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - name: Install SAM CLI
        run: script/github-action-awscli-installation
      - name: Run Deployment Script
        run: |
          echo "Deploying to beta stage for canary that talks to gamma Chime client"
          # Uses Chime Client
          npm run deploy -- -b chime-sdk-demo-beta-canary -s chime-sdk-demo-beta-canary -o chime-sdk-demo-beta-canary -i eu-south-1 -p $GAMMA_CHIME_SERVICE_PRINCIPAL -c $GAMMA_CHIME_ENDPOINT -u false -t -l
          npm run deploy -- -b chime-sdk-meeting-readiness-checker-beta-canary -s chime-sdk-meeting-readiness-checker-beta-canary -a meetingReadinessChecker -c $GAMMA_CHIME_ENDPOINT -u false -t -l
  
  deploy_chime_sdk_meetings_beta_demo:
    name: Beta - Deploy the Serverless Meeting Demos that talks to ChimeSDKMeetings Client
    runs-on: ubuntu-latest
    env:
        AWS_DEFAULT_REGION: us-east-1
        AWS_DEFAULT_OUTPUT: text
        NAME: ChimeSDKMeetingsBeta
        GAMMA_CHIME_ENDPOINT_US_EAST_1: ${{secrets.GAMMA_CHIME_ENDPOINT_US_EAST_1}}
        GAMMA_CHIME_ENDPOINT_EU_CENTRAL_1: ${{secrets.GAMMA_CHIME_ENDPOINT_EU_CENTRAL_1}}
        GAMMA_CHIME_ENDPOINT_AP_SOUTHEAST_1: ${{secrets.GAMMA_CHIME_ENDPOINT_AP_SOUTHEAST_1}}
        GAMMA_CHIME_SERVICE_PRINCIPAL: ${{secrets.GAMMA_CHIME_SERVICE_PRINCIPAL}}
    steps:
      - name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.BETA_AWS_ACCESS_KEY }}
          aws-secret-access-key: ${{ secrets.BETA_AWS_SECRET_ACCESS_KEY }}
          aws-region: us-east-1
      - name: Checkout Package
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - name: Install SAM CLI
        run: script/github-action-awscli-installation
      - name: Run Deployment Script
        run:  |
          echo "Deploying to beta stage for canary that talks to gamma ChimeSDKMeetings client"
          # Uses ChimeSDKMeetings client
          npm run deploy -- -r us-east-1 -b chime-sdk-meetings-demo-beta-canary-us-east-1 -s chime-sdk-meetings-demo-beta-canary-us-east-1 -p $GAMMA_CHIME_SERVICE_PRINCIPAL -c $GAMMA_CHIME_ENDPOINT_US_EAST_1 -t -l
          npm run deploy -- -r eu-central-1 -b chime-sdk-meetings-demo-beta-canary-eu-central-1 -s chime-sdk-meetings-demo-beta-canary-eu-central-1 -p $GAMMA_CHIME_SERVICE_PRINCIPAL -c $GAMMA_CHIME_ENDPOINT_EU_CENTRAL_1 -t -l
          npm run deploy -- -r ap-southeast-1 -b chime-sdk-meetings-demo-beta-canary-ap-southeast-1 -s chime-sdk-meetings-demo-beta-canary-ap-southeast-1 -p $GAMMA_CHIME_SERVICE_PRINCIPAL -c $GAMMA_CHIME_ENDPOINT_AP_SOUTHEAST_1 -t -l
  
  deploy_chime_gamma_demo:
    name: Gamma - Deploy the Serverless Meeting and Meeting Readiness Checker Demos that talks to Chime Client
    runs-on: ubuntu-latest
    env:
        AWS_DEFAULT_REGION: us-east-1
        AWS_DEFAULT_OUTPUT: text
        NAME: ChimeGamma
    steps:
      - name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.GAMMA_AWS_ACCESS_KEY }}
          aws-secret-access-key: ${{ secrets.GAMMA_AWS_SECRET_ACCESS_KEY }}
          aws-region: us-east-1
      - name: Checkout Package
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - name: Install SAM CLI
        run: script/github-action-awscli-installation
      - name: Run Deployment Script
        run:  |
          echo "Deploying to gamma stage for canary that talks to prod Chime client"
          # Uses Chime Client
          npm run deploy -- -b chime-sdk-demo-gamma-canary -s chime-sdk-demo-gamma-canary -o chime-sdk-demo-gamma-canary -u false -i eu-south-1 -t -l
          npm run deploy -- -b chime-sdk-meeting-readiness-checker-gamma-canary -s chime-sdk-meeting-readiness-checker-gamma-canary -a meetingReadinessChecker -u false -t -l
  
  deploy_chime_sdk_meetings_gamma_demo:
    name: Gamma - Deploy the Serverless Meeting Demos that talks to ChimeSDKMeetings Client
    runs-on: ubuntu-latest
    env:
        AWS_DEFAULT_REGION: us-east-1
        AWS_DEFAULT_OUTPUT: text
        NAME: ChimeSDKMeetingsGamma
    steps:
      - name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.GAMMA_AWS_ACCESS_KEY }}
          aws-secret-access-key: ${{ secrets.GAMMA_AWS_SECRET_ACCESS_KEY }}
          aws-region: us-east-1
      - name: Checkout Package
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - name: Install SAM CLI
        run: script/github-action-awscli-installation
      - name: Run Deployment Script
        run:  |
          echo "Deploying to gamma stage for canary that talks to prod ChimeSDKMeetings client"
          # Uses ChimeSDKMeetings client
          npm run deploy -- -r us-east-1 -b chime-sdk-meetings-demo-gamma-canary-us-east-1 -s chime-sdk-meetings-demo-gamma-canary-us-east-1 -t -l
          npm run deploy -- -r us-west-2 -b chime-sdk-meetings-demo-gamma-canary-us-west-2 -s chime-sdk-meetings-demo-gamma-canary-us-west-2 -t -l
          npm run deploy -- -r eu-central-1 -b chime-sdk-meetings-demo-gamma-canary-eu-central-1 -s chime-sdk-meetings-demo-gamma-canary-eu-central-1 -t -l
          npm run deploy -- -r ap-southeast-1 -b chime-sdk-meetings-demo-gamma-canary-ap-southeast-1 -s chime-sdk-meetings-demo-gamma-canary-ap-southeast-1 -t -l
```

@ltrung could you provide any suggestion?

Reply:
I discussed with Trung and found that you faced some issue with the script running like this and hence was split out into one. Please ignore this one.